src/box/box.cc

```diff
@@ -2605,6 +2616,12 @@ relay_get_sync_on_start(struct trigger *trigger, void *event)
 	/* Already accounted. */
 	if (bit_test(&data->collected_vclock_map, id))
 		return 0;
+
+	++data->ref_count;
+	auto guard = make_scoped_guard([&] {
+		if (--data->ref_count == 0)
+			mempool_free(&sync_trigger_data_pool, data);
```

Comment:
There's a single exit from `relay_get_sync_on_start`, by `return 0` below. So it's easier to call `sync_trigger_data_unref()` there one time than introduce a guard.